tools/clang/test/HLSLFileCheck/hlsl/workgraph/ast-nodeinput.hlsl

```diff
@@ -20,7 +20,7 @@ void node01(DispatchNodeInputRecord<RECORD> input) {}
 //CHECK-NEXT: HLSLNodeObjectAttr {{0x[0-9a-fA-F]+}} <<invalid sloc>> Implicit DispatchNodeInputRecord
 //CHECK-NEXT: FieldDecl {{0x[0-9a-fA-F]+}} <<invalid sloc>> <invalid sloc> implicit h 'int'
 //CHECK-NEXT: CXXMethodDecl {{0x[0-9a-fA-F]+}} <<invalid sloc>> <invalid sloc> Get 'const recordtype &() const'
-//CHECK-NEXT: HLSLIntrinsicAttr {{0x[0-9a-fA-F]+}} <<invalid sloc>> Implicit "op" "ExtractRecordStructFromArray" 18
+//CHECK-NEXT: HLSLIntrinsicAttr {{0x[0-9a-fA-F]+}} <<invalid sloc>> Implicit "subscript" "DefaultSubscript" 0
```

Comment:
Just a comment, I don't think any change is necessary at this point:

At first, I was concerned that there may be a problem generating a subscript operation without an index parameter, but none of the code appears to depend on the index parameter for a `DefaultSubscript` until after `LowerRecordAccessToGetNodeRecordPtr` eliminates it.

Still, it feels a little odd to be generating a DefaultSubscript function without an index parameter.